src/Microsoft.DocAsCode.Common/CommandUtility.cs

```diff
@@ -116,21 +117,35 @@ public static int RunCommand(CommandInfo commandInfo, StreamWriter stdoutWriter
         public static bool ExistCommand(string commandName)
         {
             int exitCode;
-            if (Environment.OSVersion.Platform == PlatformID.Unix || Environment.OSVersion.Platform == PlatformID.MacOSX)
+            var errorStream = new MemoryStream();
+            var outputStream = new MemoryStream();
+            using (var outputWriter = new StreamWriter(outputStream))
+            using (var errorWriter = new StreamWriter(errorStream))
             {
-                exitCode = RunCommand(new CommandInfo
+                if (Environment.OSVersion.Platform == PlatformID.Unix || Environment.OSVersion.Platform == PlatformID.MacOSX)
                 {
-                    Name = "type",
-                    Arguments = commandName
-                }, timeoutInMilliseconds: 1000);
-            }
-            else
-            {
-                exitCode = RunCommand(new CommandInfo
+                    exitCode = RunCommand(new CommandInfo
+                    {
+                        // type is a bash command, hence should be an argument to 'bash'
+                        Name = "bash",
+                        Arguments = $"-c \"type {commandName}\""
+                    }, outputWriter, errorWriter, timeoutInMilliseconds: 1000);
+                }
+                else
                 {
-                    Name = "where",
-                    Arguments = commandName
-                }, timeoutInMilliseconds: 1000);
+                    exitCode = RunCommand(new CommandInfo
+                    {
+                        Name = "where",
+                        Arguments = commandName
+                    }, outputWriter, errorWriter, timeoutInMilliseconds: 1000);
+                }
+
+                String resultError = System.Text.Encoding.UTF8.GetString(errorStream.ToArray(), 0, (int)errorStream.Length);
+                String resultOutput = System.Text.Encoding.UTF8.GetString(outputStream.ToArray(), 0, (int)outputStream.Length);
+                if (!string.IsNullOrEmpty(resultError) || !string.IsNullOrEmpty(resultOutput))
+                {
+                    Logger.LogWarning($"RunCommand exitCode: {exitCode}, output: \"{resultOutput}\", error: \"{resultError}\"");
```

Comment:
Every time this method gets called, it always log a warning as either error or output stream triggers a warning? How about letting the caller decide whether or not logging anything? For this method, it returns the error and output string..

Reply:
https://github.com/dotnet/docfx/pull/2092 to log in GitUtility when command not found